src/lib/y2partitioner/dialogs/partition_resize.rb

```diff
@@ -0,0 +1,321 @@
+    # Dialog to set new partition size
+    class PartitionResize < CWM::Dialog
+      # Constructor
+      #
+      # @param partition [Y2Storage::Partition] partition to resize
+      # @param resize_info [Y2Storage::ResizeInfo] partition resize info
+      def initialize(partition, resize_info)
+        textdomain "storage"
+
+        @partition = partition
+        @resize_info = resize_info
+        detect_space_info
+      end
+
+      # @macro seeDialog
+      def title
+        # TRANSLATORS: dialog title, where %{name} is the name of a partition (e.g., /dev/sda1)
+        format(_("Resize Partition %{name}"), name: partition.name)
+      end
+
+      # @macro seeDialog
+      def contents
+        HVSquash(
+          VBox(
+            SizeSelector.new(partition, resize_info),
+            size_info
+          )
+        )
+      end
+
+      # @macro seeDialog
+      def run
+        res = super
+
+        # TODO: check mount
+
+        res
+      end
+
+      # @macro seeDialog
+      # Necessary to mimic wizard dialog layout and behaviour
+      def should_open_dialog?
+        true
+      end
+
+    private
+
+      # @return [Y2Storage::Partition]
+      attr_reader :partition
+
+      # @return [Y2Storage::ResizeInfo]
+      attr_reader :resize_info
+
+      # @return [Y2Storage::SpaceInfo]
+      attr_reader :space_info
+
+      def detect_space_info
+        return unless formatted_partition? && committed_partition? && !swap_partition?
+        @space_info = partition.filesystem.detect_space_info
+      end
+
+      # Whether the partition is formatted
+      #
+      # @return [Boolean]
+      def formatted_partition?
+        partition.formatted?
+      end
+
+      # Whether the partition exists on disk
+      #
+      # @return [Boolean] true if the partition exists on disk; false otherwise.
+      def committed_partition?
+        system = DeviceGraphs.instance.system
+        partition.exists_in_devicegraph?(system)
+      end
+
+      # Whether the partition is for swap
+      #
+      # @return [Boolean]
+      def swap_partition?
+        partition.id.is?(:swap)
+      end
+
+      # Disk size in use
+      #
+      # @note This value only makes sense if the partition is formatted and committed.
+      #
+      # @return [Y2Storage::Disksize, nil] nil if it is not possible to detect its
+      #   space info.
+      def used_size
+        return nil if space_info.nil?
+        space_info.used
+      end
+
+      # Widgets to show size info of the partition (current and used sizes)
+      #
+      # @note Used size is only shown if space info can be detected.
+      def size_info
+        widgets = []
+        widgets << current_size_info
+        widgets << used_size_info unless space_info.nil?
+        VBox(*widgets)
+      end
+
+      # Widget for current size
+      def current_size_info
+        size = partition.size.to_human_string
+        Left(Label(format("Current size: %{size}", size: size)))
```

Comment:
Maybe the usual comments for translators just in case they feel tempted to translate the `%{size}` part? The translators are more used to the classic '%s' than to named placeholders.